hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/operations/CustomSStoreOperationTest.java

```diff
@@ -0,0 +1,162 @@
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+import com.hedera.node.app.service.contract.impl.exec.FeatureFlags;
+import com.hedera.node.app.service.contract.impl.exec.operations.CustomSStoreOperation;
+import com.hedera.node.app.service.contract.impl.exec.utils.FrameUtils;
+import com.hedera.node.app.service.contract.impl.infra.StorageAccessTracker;
+import com.hedera.node.app.service.contract.impl.state.ProxyWorldUpdater;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.apache.tuweni.units.bigints.UInt256;
+import org.hyperledger.besu.evm.EVM;
+import org.hyperledger.besu.evm.account.Account;
+import org.hyperledger.besu.evm.frame.ExceptionalHaltReason;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+import org.hyperledger.besu.evm.gascalculator.GasCalculator;
+import org.hyperledger.besu.evm.operation.Operation;
+import org.hyperledger.besu.evm.operation.SStoreOperation;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class CustomSStoreOperationTest {
+    private static final Bytes A_STORAGE_KEY = Bytes32.fromHexString("0x1234");
+    private static final Bytes A_STORAGE_VALUE = Bytes32.fromHexString("0x5678");
+
+    @Mock
+    private FeatureFlags featureFlags;
+
+    @Mock
+    private SStoreOperation delegate;
+
+    @Mock
+    private EVM evm;
+
+    @Mock
+    private MessageFrame frame;
+
+    @Mock
+    private GasCalculator gasCalculator;
+
+    @Mock
+    private ProxyWorldUpdater proxyWorldUpdater;
+
+    @Mock
+    private StorageAccessTracker accessTracker;
+
+    @Mock
+    private Account account;
+
+    private SStoreOperation realSStoreOperation;
+
+    private CustomSStoreOperation subject;
+
+    @BeforeEach
+    void setUp() {
+        realSStoreOperation = new SStoreOperation(gasCalculator, FRONTIER_MINIMUM);
+        subject = new CustomSStoreOperation(featureFlags, delegate);
+    }
+
+    @Test
+    void propagatesUnsuccessfulResults() {
+        final var haltResult = new Operation.OperationResult(123, ExceptionalHaltReason.INSUFFICIENT_STACK_ITEMS);
+        given(delegate.execute(frame, evm)).willReturn(haltResult);
+
+        final var result = subject.execute(frame, evm);
+
+        assertSame(haltResult, result);
+    }
+
+    @Test
+    void doesNoTrackingIfSidecarDisabled() {
+        final var successResult = new Operation.OperationResult(123, null);
+
+        given(delegate.execute(frame, evm)).willReturn(successResult);
+        given(frame.getStackItem(0)).willReturn(A_STORAGE_KEY);
+
+        final var result = subject.execute(frame, evm);
+
+        assertSame(successResult, result);
+
+        verify(featureFlags).isSidecarEnabled(frame, CONTRACT_STATE_CHANGE);
+        verifyNoMoreInteractions(frame);
```

Comment:
Which part of this test causes the read that would trigger the tracking of the read value if the sidecar feature was enabled?

Reply:
That only happens in the `tracksReadValueOnSuccess()` test. Or am I misunderstanding?